server/src/main/java/io/crate/statistics/TransportAnalyzeAction.java

```diff
@@ -211,11 +199,9 @@ static Stats createTableStats(Samples samples, List<Reference> primitiveColumns)
 
     private CompletableFuture<Samples> fetchSamples(RelationName relationName, List<Reference> columns) {
         FutureActionListener<FetchSampleResponse, Samples> listener = new FutureActionListener<>(FetchSampleResponse::samples);
-        List<DiscoveryNode> nodesOn41OrAfter = StreamSupport.stream(clusterService.state().nodes().spliterator(), false)
-            .filter(x -> x.getVersion().onOrAfter(Version.V_4_1_0))
-            .collect(Collectors.toList());
+        DiscoveryNodes discoveryNodes = clusterService.state().nodes();
         MultiActionListener<FetchSampleResponse, ?, FetchSampleResponse> multiListener = new MultiActionListener<>(
-            nodesOn41OrAfter.size(),
+            discoveryNodes.getSize(),
```

Comment:
Could we split these changes out into a dedicated PR?

Reply:
https://github.com/crate/crate/pull/13052